pkg/envoy/ads/stream.go

```diff
@@ -172,7 +172,20 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
 			log.Info().Msgf("Discovery request <%s> for resources (%v) from Envoy UID=<%s> with Nonce=%s",
 				xdsShortName, discoveryRequest.ResourceNames, proxy.GetPodUID(), discoveryRequest.ResponseNonce)
 
-			err := s.sendResponse(mapset.NewSetWith(typeURL), proxy, &server, &discoveryRequest, s.cfg)
+			var xdsUpdatePaths mapset.Set
+			switch typeURL {
+			case envoy.TypeWildcard:
+				xdsUpdatePaths = mapset.NewSetWith(
+					envoy.TypeCDS,
+					envoy.TypeEDS,
+					envoy.TypeLDS,
+					envoy.TypeRDS,
+					envoy.TypeSDS)
```

Comment:
Does order matter here? If it does, does mapset gaurantee ordering (I guess not)?

Reply:
(1) i think order is not guaranteed (i think it uses internal map?) and (2) order here doesn't matter

Order is however guaranteed when calling `sendResponse` over [here](https://github.com/openservicemesh/osm/pull/2938/files#diff-7a1f450fbabb0088828083ed90ad3ccc49ed4edd9ae6f58ad5bdcdd158b2dda8R62)